datasource-decorator-spring-boot-autoconfigure/src/main/java/com/github/gavlyukovskiy/micrometer/DataSourceMetricsJdbcEventListener.java

```diff
@@ -34,13 +34,13 @@ public void onBeforeGetConnection(ConnectionInformation connectionInformation) {
     public void onAfterGetConnection(ConnectionInformation connectionInformation, SQLException e) {
         String dataSourceName = p6SpyDataSourceNameResolver.resolveDataSourceName(connectionInformation.getDataSource());
         DataSourceMetricsHolder metrics = dataSourceMetricsBinder.getMetrics(dataSourceName);
-        metrics.afterAcquireConnection(connectionInformation.getConnectionId(), connectionInformation.getTimeToGetConnectionNs(), TimeUnit.NANOSECONDS, e);
+        metrics.afterAcquireConnection(connectionInformation, connectionInformation.getTimeToGetConnectionNs(), TimeUnit.NANOSECONDS, e);
```

Comment:
Does `getTimeToGetConnectionNs` contain true nanosecond resolution or is it a pre-scaled version of a millisecond resolution timer? If the former, cool! If the latter, there is no need to do such pre-scaling in your app, as Micrometer will scale to the appropriate base unit of time for each backend based on the source unit you provide at instrumentation time.

Reply:
Yes, this listener is from p6spy that is measuring in nanoseconds. Another one - DataSourceMetricsMethodExecutionListener from datasource-proxy has only milliseconds